Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -2353,29 +2353,131 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
 
     void vectorMax(SIMDInfo simdInfo, FPRegisterID left, FPRegisterID right, FPRegisterID dest)
     {
-        UNUSED_PARAM(right); UNUSED_PARAM(dest);
-        if (scalarTypeIsFloatingPoint(simdInfo.lane))
-            UNUSED_PARAM(left);
-        else {
-            ASSERT(simdInfo.signMode != SIMDSignMode::None);
-            if (simdInfo.signMode == SIMDSignMode::Signed)
-                UNUSED_PARAM(left);
-            else
-                UNUSED_PARAM(left);
+        ASSERT(simdInfo.signMode != SIMDSignMode::None);
+
+        switch (simdInfo.lane) {
+        case SIMDLane::i8x16:
+            if (supportsAVX()) {
+                if (simdInfo.signMode == SIMDSignMode::Signed)                
+                    m_assembler.vpmaxsb_rr(right, left, dest);
+                else
+                    m_assembler.vpmaxub_rr(right, left, dest);
+            } else {
+                if (left != dest)
+                    m_assembler.movapd_rr(left, dest);
+                if (simdInfo.signMode == SIMDSignMode::Signed) {
+                    if (supportsFloatingPointRounding())
+                        m_assembler.pmaxsb_rr(right, dest);
+                    else
+                        RELEASE_ASSERT_NOT_REACHED();
+                } else
+                    m_assembler.pmaxub_rr(right, dest);
+            }
+            return;
+        case SIMDLane::i16x8:
+            if (supportsAVX()) {
+                if (simdInfo.signMode == SIMDSignMode::Signed)                
+                    m_assembler.vpmaxsw_rr(right, left, dest);
+                else
+                    m_assembler.vpmaxuw_rr(right, left, dest);
+            } else {
+                if (left != dest)
+                    m_assembler.movapd_rr(left, dest);
+                if (simdInfo.signMode == SIMDSignMode::Signed)
+                    m_assembler.pmaxsw_rr(right, dest);
+                else {
+                    if (supportsFloatingPointRounding())
```

Comment:
Add `supportsSSE4_1()` and use it instead. `supportsFloatingPointRounding` should be used only for rouding.